### PR TITLE
Add Sophus

### PR DESCRIPTION
**Pull request by versatran01**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by versatran01** [comment]
@conda-forge-admin, please ping team

**Comment by versatran01** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by fhill122** [comment]
there is a typo in cmake args, should be "-DBUILD_SOPHUS_TESTS=OFF" (missing S)

File: recipes/sophus/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/strasdat/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 635dc536e7768c91e89d537608226b344eef901b51fbc51c9f220c95feaa0b54
+
+build:
+  number: 0
+  skip: true  # [win and vc<14]
```
Comment (marcelotrevisani) [inline recipes/sophus/meta.yaml:14]:
```suggestion
  skip: true  # [win]
```

**Comment by versatran01** [comment]
@conda-forge-admin, please ping team, this PR is ready.

File: recipes/sophus/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  
+  host:
+    - eigen
+  
+  run: 
+    - eigen
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/sophus/types.hpp  # [unix]
```
Comment (carterbox) [inline recipes/sophus/meta.yaml:31]:
```suggestion
  commands:
    - test -f ${PREFIX}/include/sophus/types.hpp                # [unix]
    - test -f ${PREFIX}/share/sophus/cmake/SophusTargets.cmake  # [unix]
```

File: recipes/sophus/build.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+mkdir -p build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+      -DCMAKE_PREFIX_PATH=${PREFIX} \
+      -DBUILD_SOPHUS_EXAMPLES=OFF \
+      -DBUILD_SOPHUS_TESTS=OFF \
+      -DSOPHUS_USE_BASIC_LOGGING=ON \
+      ..
+make -j${CPU_COUNT}
+make install
```
Comment (carterbox) [inline recipes/sophus/build.sh:10]:
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} \
      -DCMAKE_PREFIX_PATH=${PREFIX} \
      -DBUILD_SOPHUS_EXAMPLES=OFF \
      -DBUILD_SOPHUS_TESTS=ON \
      -DSOPHUS_USE_BASIC_LOGGING=ON \
      ..
make -j${CPU_COUNT}
make test
make install
```
You should build and run at least one test to check that the conda-forge compilers are compatible with the source and compiler args.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/sophus/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/strasdat/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 635dc536e7768c91e89d537608226b344eef901b51fbc51c9f220c95feaa0b54
+
+build:
+  number: 0
+  skip: true  # [win]
```
[the remark this answers is not in the document]
Comment (versatran01) [reply recipes/sophus/meta.yaml:14]: https://github.com/strasdat/Sophus/issues/358
Here the maintainer says to leave out win for now until ci is setup

**Comment by versatran01** [comment]
@conda-forge-admin, please ping team

File: recipes/sophus/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make  # [not win]
+  
+  host:
+    - eigen
+  
+  run: 
+    - eigen
```
[the remark this answers is not in the document]
Comment (carterbox) [reply recipes/sophus/meta.yaml:27]: You marked this as resolved, but nothing changed. Why not include the minimum Eigen version specification?

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by zzz6519003** [comment]
@conda-forge-admin is a bot? cool

**Pull request merged**